Restore lost functionality

RoystonS opened the pull request:
This patch restores various bits of functionality which were lost in a previous pull request.
- Restore timeouts
- Restore unit tests for JazzConfiguration, and a switch from Cloneable to a copy constructor that had also been lost
- Fix NPEs in config parsing
- Failed SCM polls no longer trigger builds off

Royston-Shufflebotham-i2 committed:
Fix NPEs by using fixEmptyAndTrim instead of fixEmpty().trim()
Encode empty password as null.

Royston-Shufflebotham-i2 committed:
Reinstated timeout configuration support.

Restored a couple of patches which were lost after an earlier merge.
Timeout was originally merged in at commit 973453bf17c802deb3c31093fde9b6058160785f
Restored JazzConfiguration copy constructor from commit 7e1b287fe841a369ddec374d87212ce477d496a1
Restored (+ updated) JazzConfigurationTests.

buildhive commented:
[Jenkins » rtc-plugin #25](https://buildhive.cloudbees.com/job/jenkinsci/job/rtc-plugin/25/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

Royston-Shufflebotham-i2 committed:
Fixed bug where failed SCM polls caused builds to trigger.

If an SCM poll failed in compare(), the error was caught, and a fake changeset
returned to the caller.  This then triggered a build to happen because it thought a
change had been detected.  Reverted.

buildhive commented:
[Jenkins » rtc-plugin #26](https://buildhive.cloudbees.com/job/jenkinsci/job/rtc-plugin/26/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[Jenkins » rtc-plugin #27](https://buildhive.cloudbees.com/job/jenkinsci/job/rtc-plugin/27/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

Royston-Shufflebotham-i2 committed:
Restore fixes lost by a previous merge.

A number of fixes were applied by @deluan in commit
ba2282a01b61bb69084e553f50201cef2dc9f0ab but were subsequently lost
amongst the merges in pull request #7.

buildhive commented:
[Jenkins » rtc-plugin #28](https://buildhive.cloudbees.com/job/jenkinsci/job/rtc-plugin/28/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)
